### PR TITLE
feat: Bb 150 implement exporting json to nmp

### PR DESCRIPTION
**Pull request by GDamaso**
# Description

You can now download an nmp file by clicking the export button. So far, only FarmName and Year variable are available for edditing inside BetterBerries. It will be expanded in the future.

## Type of changes

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

Export functionality demo:

ExportDemo

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-sustainment-capstone-2024-206-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-sustainment-capstone-2024-206-backend.apps.silver.devops.gov.bc.ca/api/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/Commons/CustomLink/CustomLink.style.ts [hunk 1]
```diff
@@ -32,12 +31,11 @@ const StyledLinkContainer = styled.div<StyledLinkProps>`
     align-items: center;
     justify-content: center;
     color: inherit;
+    padding: 20px 30px;
```
Comment (GDamaso) [inline frontend/src/Commons/CustomLink/CustomLink.style.ts:34]: Adjusted CustomLinks clickable area

**Review by GDamaso — COMMENTED** [review]
Self Reviewed :ok_man:

File: frontend/src/Commons/Forms/FormHeader/FormHeader.tsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ interface FormHeaderProps {
 
 const FormHeader = ({ inputModule, handleFormState }: FormHeaderProps) => {
   const handleToggle = () => {
-    handleFormState();
+    handleFormState(inputModule.id.toString());
```
Comment (kcaparas1630) [inline frontend/src/Commons/Forms/FormHeader/FormHeader.tsx:30]: 🚀

File: frontend/src/Interface/nmpInterface.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      ];
+      SeparatedSolidManures: [];
+      ManureStorageSystems: [];
+    },
+  ];
+  LastAppliedFarmManureId: null;
+  NMPReleaseVersion: number;
+}
+
+export default nmpInterface;
```
Comment (kcaparas1630) [inline frontend/src/Interface/nmpInterface.tsx:150]: the plugin was really helpful building this 🚀

File: frontend/src/Views/Export/ExportPage.tsx [hunk 4]
```diff
@@ -13,8 +13,20 @@ import { StyledContent, StyledLandingContainer } from './ExportPage.styles.ts';
 
 const ExportPage: FC = () => {
   const downloadFile = () => {
-    localStorage.clear();
-    console.log('Downloading');
+    // localStorage.clear();
```
Comment (kcaparas1630) [inline frontend/src/Views/Export/ExportPage.tsx:16]: is it supposed to be not cleared? Whenever I download it, and create a new calculation, the loaded  values are still there. When it should be an empty string.

**Review by kcaparas1630 — COMMENTED** [review]
There is not much I can say here, except for great work! I believe that you are improving at a faster rate, and I'm amazed at what you're doing. Keep up the great work.

I've tried and tested, and I didn't encounter any problem except for the input fields bug that I've mentioned in the comment.

File: frontend/src/Interface/nmpInterface.tsx (hunk 3 above)
Comment (GDamaso) [reply frontend/src/Interface/nmpInterface.tsx:150]: Actually did this with macros hehe :smile:

File: frontend/src/Views/Export/ExportPage.tsx (hunk 4 above)
Comment (GDamaso) [reply frontend/src/Views/Export/ExportPage.tsx:16]: Actually, I believe it should be cleared when we click on new calculation. I'll change that, forgot to do it in the end. Nice catch!

File: frontend/src/Interface/nmpInterface.tsx (hunk 3 above)
Comment (kcaparas1630) [reply frontend/src/Interface/nmpInterface.tsx:150]: Haha I was thinking of plugins of vscode

File: frontend/src/Views/Export/ExportPage.tsx (hunk 4 above)
Comment (GDamaso) [reply frontend/src/Views/Export/ExportPage.tsx]: Had to change the link to a Button since it's now doing stuff aside from moving the page :ok_hand:

**Review by kcaparas1630 — APPROVED** [review]
Good job! 🚀

**Pull request merged**